### PR TITLE
AddonLifecycle Performance Enhancement

### PR DESCRIPTION
**Pull request by MidoriKami**
Per Aers' somewhat obvious question in hindsight... ("Why do you need to hook it before its requested?"), I have tweaked it slightly to instead store the address of the ReceiveEvent function, and now only hooks it when it is requested by someone.

### CODE REVIEW

**Comment by MidoriKami** [comment]
Marked as draft for now, as FFXIV is currently down for All Worlds Maintenance.

**Comment by MidoriKami** [comment]
Ready for merge, tested and verified working.

https://github.com/goatcorp/Dalamud/assets/9083275/f3a57e31-3d87-4c6a-bc3f-65f931d51c02

This should also help mitigate an issue reported on Linux with reloaded causing noticeable game lag, as hooks will only be generated when requested by a plugin.

File: Dalamud/Game/Addon/Lifecycle/AddonLifecycleAddressResolver.cs [hunk 1]
```diff
@@ -63,6 +65,6 @@ protected override void Setup64Bit(ISigScanner sig)
         this.AddonUpdate = sig.ScanText("FF 90 ?? ?? ?? ?? 40 88 AF");
         this.AddonOnRequestedUpdate = sig.ScanText("FF 90 98 01 00 00 48 8B 5C 24 30 48 83 C4 20");
         this.AddonOnRefresh = sig.ScanText("48 89 5C 24 08 57 48 83 EC 20 41 8B F8 48 8B DA");
-        this.AtkEventListener = sig.GetStaticAddressFromSig("4C 8D 3D ?? ?? ?? ?? 49 8D 8E");
+        this.AtkUnitBase = (AtkUnitBase*)sig.GetStaticAddressFromSig("48 8D 05 ?? ?? ?? ?? 48 89 01 48 8D 05 ?? ?? ?? ?? 4C 89 41 28");
```
Comment (KazWolfe) [inline Dalamud/Game/Addon/Lifecycle/AddonLifecycleAddressResolver.cs:68]: Can these sigs be moved to CS proper?

Comment (MidoriKami) [reply Dalamud/Game/Addon/Lifecycle/AddonLifecycleAddressResolver.cs:68]: The callsite sigs can not, as those are for very specific code points.

AddonOnRequestedUpdate is likely already in clientstructs. It's one of the virtual functions for AtkUnitManager.

AtkUnitBase's vtable might already be in clientstructs now that I think about it.

Comment (MidoriKami) [reply Dalamud/Game/Addon/Lifecycle/AddonLifecycleAddressResolver.cs:68]: It doesn't seem that clientstructs has a address for AtkUnitBase's vtablen from what I can tell

Comment (MidoriKami) [reply Dalamud/Game/Addon/Lifecycle/AddonLifecycleAddressResolver.cs]: I was only able to remove AtkUnitBase, the only other one we can remove is OnRefresh, but currently ClientStructs has the wrong function signature, and that will cause issues, so it's on hold for now.

**Comment by MidoriKami** [comment]
https://github.com/aers/FFXIVClientStructs/pull/926 has been merged into ClientStructs

When this is merged into Dalamud, I can remove the `AddonOnRefresh` sig

**Pull request merged**